maven-release-api/src/main/java/org/apache/maven/shared/release/config/ReleaseDescriptor.java

```diff
@@ -426,6 +426,13 @@
      */
     String getProjectVersionPolicyId();
 
+    /**
+     * Get the (optional) config for the VersionPolicy implementation used to calculate the project versions.
+     *
+     * @return The parsed XML of the provided config (an instance of XmlPlexusConfiguration) or null.
+     */
+    Object getProjectVersionPolicyConfig();
+
```

Comment:
Does not really make sense to tie it to XML, consider someone is using this outside of Maven realm and uses real objects throughout. Why not create a simple interface `VersionPolicyConfig` which has a `#serialize()` method. For the PlexusDom we can create an impl which will delegate to `#toString()`, thus we can cleanly separate between Maven Release and Maven Core usage. WDYT?`

Reply:
Given that this is a maven plugin I do not understand the `using this outside of Maven realm`.
The ProjectVersionConfig has been typed as a `DOM` and thus it is always an XML.
I find this a perfectly acceptable constraint for this kind of config.

Reply:
Also, I do not fully understand what you have in mind.

Reply:
I don't agree with that. The Maven Release API is *not* the plugin. The Maven Release Plugin is a user of Maven Release. What if someone wants to write a custom plugin which uses the Maven Release API w/o our plugin?

Reply:
Ok, I'll have a look what I can do to improve this. I like the `VersionPolicyConfig` proposal.

Reply:
Trying to make it cleaner.
Observations so far:
- In the Modello of the `ModelloReleaseDescriptor` the config needs to be of type `DOM`. If it isn't then I have to wrap the config into CDATA which makes it a terrible developer experience.
- This generated class contains: `private Object projectVersionPolicyConfig;` and the related getter and setter that go with this also use `Object` as the class. 
- I checked the Modello code and I do not see a way to let it return a different type (like a `VersionPolicyConfig`).
- These getters and setters must be the same in the `ReleaseDescriptor` interface (i.e. also `Object`.

This is the code you mentioned.
If I simply change the return type then this results in a conflict with the generated code. I'm unable to change the generated code regarding this type.

Note that the other reason for the way I have it now was that at some point the config is also stored in a Property ... which only accepts String.

Please advise. I'm a bit stuck.

Reply:
Alright, thanks for your investigation" Let me double check this tomorrow, maybe there is currently no better way to solve

Reply:
Have you been able to find a better way? 
I'm happy to implement it if it is possible.

Reply:
Haven't looked, was busy with Resolver. Will dedicate some time to it this evening.

Reply:
`org.apache.maven.shared.release.config.ModelloReleaseDescriptor` does config any general purpose config element and nothing as `DOM`. Can you clarify what you are referring to?

Reply:
In the `mdo` file you can specify a bunch of types and you can specify (xml) nesting.

If you cannot specify the entire thing in there (like in this case because all custom Version Policies will have currently unknown config options) you either end up with
- it being a `string` which means you have to CDATA the config which results in a very bad developer experience, but it does allow you to have any kind of config format you like (xml, json, yaml, ...) .
- it being a `dom` which makes it an XML without schema. The effect is that an IDE can only assist you in making sure it is an XML (which is not validated other than having the open and close tags in sync). And the effect is that the generate Modello code says it is an `Object` which causes what I tried to describe above.

Reply:
Ah OK, I was confused by your statement because the modello model did not change. Can you update your findings in that regard and I will check again?

Reply:
I do not understand right now what you would like me to update.

Reply:
I wonder if this is always going to be `XmlPlexusConfiguration` why not make it `XmlPlexusConfiguration`? Too much overhead? I mean the Javadoc is clear on this, no?

Reply:
I put in the JavaDoc it is really this type. 
It is the code generated by Modello that dictates to to be `Object`. 

I have tried changing it to `XmlPlexusConfiguration` but the current Modello code generator won't let me.

The only way to actually make it an `XmlPlexusConfiguration` is by changing (probably) this: https://github.com/codehaus-plexus/modello/blob/master/modello-plugins/modello-plugin-java/src/main/java/org/codehaus/modello/plugin/java/JavaModelloGenerator.java#L1368

Reply:
Sigh, there is even an open TODO regarding this in Modello code. So, I guess we need to swallow the `Object`.